GoogleSignIn/Sources/GIDSignIn.m

```diff
@@ -996,24 +997,19 @@ - (void)assertValidPresentingViewController {
 }
 
 #if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
-// Asserts the requested scopes are valid.
 - (void)assertValidScopes:(NSArray<NSString *> *)scopes {
-  NSDictionary *scopeToClassMapping = @{
-    kAccountDetailTypeAgeOver18Scope : [GIDVerifyAccountDetail class],
-  };
-  NSMutableString *errorMessage =
+  GIDRestrictedScopesRegistry *registry = [[GIDRestrictedScopesRegistry alloc] init];
+  NSDictionary<NSString *, Class> *restrictedScopesMapping = [registry restrictedScopeToClassMappingInSet:[NSSet setWithArray:scopes]];
+
+  if (restrictedScopesMapping.count > 0) {
+    NSMutableString *errorMessage =
     [NSMutableString stringWithString:@"The following scopes are not supported in the 'addScopes' flow. "
                                        "Please use the appropriate classes to handle these:\n"];
-  Boolean unsupportedScopeFound = NO;
-
-  for (NSString *scope in scopes) {
-    Class scopeClass = scopeToClassMapping[scope];
-    if (scopeClass) {
-      unsupportedScopeFound = YES;
-      [errorMessage appendFormat:@"%@ -> %@\n", scope, NSStringFromClass(scopeClass)];
+    for (NSString *restrictedScope in restrictedScopesMapping) {
```

Comment:
You can enumerate keys and values at the same time: https://developer.apple.com/documentation/foundation/nsdictionary/1414570-enumeratekeysandobjectsusingbloc

Would remove the need for the line below.

Reply:
Cool! Incorporated this in, thanks.